### DISCUSSION TITLE
The Wholesome Hacienda Overhaul v2.0

### DISCUSSION THREAD

**Discussion by Taludas** [Taludas/WholesomeHaciendaHaul#8] [Ideas]
![Thumbnail](https://user-images.githubusercontent.com/64583643/192149385-1cfd642f-21e5-4d6e-96e2-8d794a07bd25.png)

Update for GU16.

The Mod is now complete restructured and intended to be used with the Mod Manager IModYourAnno for Tweaking of Mod Features. Modules can be disabled there or through manually editing the main assets.xml to remove certain mod modules by deleting their <Include>-Command.
If you have further wishes in customized modules through IMYA, feel free to post in the discussions!

Please refer to the changelog for a detailed overview!

<hr /><em>This discussion was created from the release <a href='https://github.com/Taludas/WholesomeHaciendaHaul/releases/tag/v2.0'>The Wholesome Hacienda Overhaul v2.0</a>.</em>

**Comment by Canderous1989** [comment]
Kann es sein, dass mit dem Düngewerk etwas nicht stimmt? Anders als früher kann ich es nicht mehr unabhängig von der Hacienda platzieren. Die entsprechende .xml Datei habe ich nicht verändert.

**Comment by Taludas** [comment]
Ja, das Düngewerk ist jetzt standardmäßig auf Vanilla Einstellungen. Du musst den iModYourAnno Mod Manager nutzen, oder die xml Datei per Hand anpassen, um den Zustand vor dem Mod wieder herzustellen.In IMYA kannst du dann ganz einfach einen Toggle nutzen, um das Modul von der Hacienda unabhängig zu machen. Also alles working as intended. Muss ich im Changelog noch Mal genauer spezifizieren. Danke für den Hinweis!
 
 
  On Mon, 19 Dec 2022 at 20:45, ***@***.***> wrote:   


Kann es sein, dass mit dem Düngewerk etwas nicht stimmt? Anders als früher kann ich es nicht mehr unabhängig von der Hacienda platzieren. Die entsprechende .xml Datei habe ich nicht verändert.

**Comment by Canderous1989** [comment]
Danke für die Antwort. Ich benutze aktuell noch den Anno Mod Loader. Was genau muss ich denn in der modularfertiliserworks.include.xml ändern? Ich stehe da gerade etwas auf dem Schlauch. Welchen Befehl muss ich löschen?

**Comment by Taludas** [comment]
Da muss irgendwo etwas von <IsHaciendaDependent> stehen. Da musst du den Wert auf 0 ändern. Dann kannst due s auch außerhalb bauen. Die andern Werte aus der alten Mod: 90sec cycle time, 1/10 extra goods on animal farms. Rest müsste vanilla passen.
 
 
  On Mon, 19 Dec 2022 at 22:09, ***@***.***> wrote:   


Danke für die Antwort. Ich benutze aktuell noch den Anno Mod Loader. Was genau muss ich denn in der modularfertiliserworks.include.xml ändern? Ich stehe da gerade etwas auf dem Schlauch. Welchen Befehl muss ich löschen?

**Comment by Canderous1989** [comment]
Mhm. Ich habe die Befehlszeile wie folgt angepasst:

    <ModOp Type="replace" GUID="24805" Path="/Values/Pausable/IsHaciendaDependent" ModOpID="FertiliserHaciendaDependent">
        <IsHaciendaDependent>0</IsHaciendaDependent>

Das Ändern von 1 auf 0 führt zwar dazu, dass ich außerhalb der Hacienda bauen kann (ging schon vorher) aber sie steht immer noch still mit dem Verweis auf den fehlenden Hacienda-Reichweite.

Meinst du eine andere Zeile?

**Comment by Taludas** [comment]
ModOp Type="replace" GUID="24805" Path="/Values/Pausable/IsHaciendaDependent" ModOpID="FertiliserHaciendaDependent"><IsHaciendaDependent>0</IsHaciendaDependent></ModOp> 
Das sollte zum gewünschten Ergebnis führen. 
 
  On Mon, 19 Dec 2022 at 22:26, ***@***.***> wrote:   


Mhm. Ich habe die Befehlszeile wie folgt angepasst:

ModOp Type="replace" GUID="24805" Path="/Values/Pausable/IsHaciendaDependent" ModOpID="FertiliserHaciendaDependent">
0


Das Ändern von 1 auf 0 führt zwar dazu, dass ich außerhalb der Hacienda bauen kann (ging schon vorher) aber sie steht immer noch still mit dem Verweis auf den fehlenden Hacienda-Anschluss.

Meinst du eine andere Zeile?

**Comment by Canderous1989** [comment]
Seltsam. Genau die Änderung habe ich ja vorgenommen. Ergebnis ist leider das oben Beschriebene.

**Comment by Taludas** [comment]
Bitte nach einer Änderung auch immer den Cache Ordner im mods Ordner löschen. Sonst lädt er die Änderungen der include.xml nicht ins Spiel. 
 
 
  On Mon, 19 Dec 2022 at 22:44, ***@***.***> wrote:   


Seltsam. Genau die Änderung habe ich ja vorgenommen. Ergebnis ist leider das oben Beschriebene.

**Comment by Canderous1989** [comment]
Das war die Lösung :)

**Comment by Nikoleagle** [comment]
Is it possible to additionally add the "old" hacienda street and pathway somewhere in build menu? New method to build those as roads is amazing, but it creates subtle cosmetic issues sometimes. When "pathway" meets "street" the latter become abruptly stopped.
![screenshot_2022-12-31-18-38-22](https://user-images.githubusercontent.com/121718136/210150233-5a10e017-6389-40be-938f-630d6a255c30.jpg)

**Comment by Taludas** [comment]
Currently you can choose from the two options by disabling the module in the main asset file. But then the other street forms are missing.If you want both you have to edit the include file from replace to addnextsibling. Usually you should have no problem with adaption of the streets. Maybe delete the old street tiles and only use new ones. They should work no problem.
 
 
  On Sat, 31 Dec 2022 at 17:45, ***@***.***> wrote:   


Is it possible to additionally add "old" hacienda street and pathway somewhere in build menu? New method to build those as roads is amazing, but it creates subtle cosmetic issues sometimes. When "pathway" meets "street" the latter become abraptly stopped

**Comment by Lamproly** [comment]
Hi Taludas! 
How would I go about deleting the mod? The only module I was using is the street one. I haven't built those streets and wanted the old system back. Deleted mod and cache, but the problem is, I can't select the vanilla hacienda streets now nor copy them. Even after verifying files, reinstalling mod loader etc. the bug persists. I also tried disabling the module first, delete cache, make a save, delete mod, delete cache, still same problem. 
The only thing fixing it is going back to a save before installing Wholesome Hacienda, but that is very far back, so not really an option. 
Any ideas how to fix it?

**Reply by Taludas** [reply]
So the problem with streets is, that they are savegame persistent. So even if you delete the streets assets by disabling the mod, verifing files etc. the modded streets will always be saved inside your savegame.
The problem coming with this is, that you can not copy those street assets anymore nor interact with them in any way. But this should not hinder you in buildinge the old tiles again. With disabling this mod, they should regain their old position in the hacienda build menu and in the module tab of the hacienda and should be fully functional.

**Reply by Lamproly** [reply]
They do appear again in the old hacienda menu, but if I click on them, the buttons only briefly highlight, but don't "attach" the streets to my cursor. Menu stays open. All other hacienda modules are buildable just fine.

**Comment by Taludas** [comment]
I'll have a look at that, and see if I can replicate the issue and see how to fix it!
 
 
  On Wed, 11 Jan 2023 at 16:57, ***@***.***> wrote:   


They do appear again in the old hacienda menu, but if I click on them, the buttons only briefly highlight, but don't "attach" the streets to my cursor. Menu stays open. All other hacienda modules are buildable just fine.

**Comment by miri4ever** [comment]
Ich hatte bis gestern noch die 1.4 und konnte dann aber die Häuser nicht mehr aufsteigen lassen und die Symbole doppelt im Menü. Die anderen Sachen konnte ich problemlos bauen. Ich fand nun das Update auf die 2.01 was aber nicht so einfach war, weil man überall nur auf die alten Versionen gerade über die Anno MOD Seite. 

Ich habe heute ein wenig gespielt und wollte meine Hacienda weiterbauen und die Objekte wie Pflanzen und Brunnen nutzen. Ich musste jetzt aber feststellen, dass es kein Menü dafür mehr gibt. Ich konnte die alten zuvor gebauten zwar kopieren, aber keine neuen setzen und auch die Mauer fehlt komplett.  Lagerhaus, Düngewerk und dann ist nicht mehr in der Anzeige mit Ornamenten oder den Mauern wie zuvor noch in 1.4. Habe alle Menüs durchsucht, kann die Sachen aber nicht mehr finden.  

Kann mir hier einer weiterhelfen ? Auch habe ich nur die Feuerwehr aber keine Polizei oder Krankenhaus zur Auswahl.

**Comment by miri4ever** [comment]
![111](https://user-images.githubusercontent.com/16743819/218167149-8abfb767-9624-44c4-bc2a-bfa27ca50f6c.jpg) I have made this image to show my view of hacienda with no ornaments, walls, police, or hospital to build. I have deleted the folder and several times replaced with the latest mod. The only time it worked was at the older version, where the ornaments split with this extra mod file at 1.41 or older. I even deleted my hacienda and rebuild new, also no change.
